### ISSUE TITLE
Color as Dimension vs Type

### ISSUE THREAD

**Issue by azaroth42** [linked-art/linked.art#191]
At the moment, we have color mapped as a Dimension -- "white" is value "FFFFFF" with unit "RGB colorspace". But if our underlying data has "white" ... do we add unintended precision by expressing it as RGB? It could easily be "FEFEFE" or any other very-close-to-white RGB color.

We could have multiple color spaces, and use (e.g.) "FFF" for a 16 bit per channel space rather than 256. Or we could use an E55 Type as classification, as per Shape.

**Comment by mdlincoln** [comment]
What if the source data uses a vocabulary term like [aat:white](http://www.getty.edu/vow/AATFullDisplay?find=red&logic=AND&note=&subjectid=300129784)?

**Comment by azaroth42** [comment]
That would lead us towards just another `P2_has_type` / `classified_as` property, I think. At the Getty, the databases I've seen with color information (in Conservation and the Museum, notably) have free text fields :/ They could even just be description statements rather than controlled vocabulary based, let alone a measurement of a dimension.

**Comment by workergnome** [comment]
Is there a good thesauri for color? Is the [AAT color facet](http://www.getty.edu/vow/AATFullDisplay?find=&logic=AND&note=&subjectid=300131647) sufficient?  They've got a lot of terms in there, based on the [ISCC–NBS system](https://en.wikipedia.org/wiki/ISCC–NBS_system).

I'm familiar with the CSS color names, (though those could also be dimensions).

**Comment by jeremytubbs** [comment]
I believe this should be a dimension value because when I implemented this for the IMA it involved a prominence value. It is definitely only an `Aboutness` value. 

I could see this as:

```
"color":  [
     {
         "id": "https://linked.art/example/value/6", 
         "type": "Concept", 
         "value": 0.5,
         "classified_as": [
             {
                 "id": "http://vocab.getty.edu/aat/300129784", 
                 "type": "Type", 
                 "label": "White"
             }
         ]
   }
]

```

I would like a way to record the percentage of color. Below is a screen grab of an example of color data the IMA was capturing... 


![image](https://user-images.githubusercontent.com/548888/52020898-ecff8400-24c0-11e9-8a4b-556c41d609e4.png)

**Comment by azaroth42** [comment]
I think the modeling difference here is between categorical types (e.g. boolean flags that do not interact with other flags ... something is not more white, it is either white or not) compared to ordinal types (there is an order, so something can be more white without being precise as to how much more white) and interval types (there is a value that can be compared to determine how much more white).

The Dimension approach is the interval type -- there is a unit (RGB) and a value (FFFFFF) that can be directly compared. This seems to introduce too much false precision.
The Type approach is categorical -- there is only a flag, such as aat:white. This seems to lack functionality.
Do we actually need the ordinal middle ground, or is that not solving anyone's problem and we can agree on one of the other two?

**Comment by natuk** [comment]
In our experience, categorical colour assignments have proved useful only when kept as simple as possible (e.g. see the colour dropdown list [here](https://www.ligatus.org.uk/decoratedpapers/frontpage?field_decoration_by_no_of_colour_tid_selective=All&field_colours_tid_selective%5B%5D=6&field_tool_tid_selective=All) ) This allows easy searching based on colour. Applications which measure colour as part of a technical process would require a dimensions structure. I do not see anything wrong with marking colour as a Feature which has a Dimension and a Type. Would that complicate the model too much?

**Comment by zeroexp** [comment]
In ImageSnippets we create a property: lio:hasColor (for simplicity) - but it could be another property from another ontology. Then use (usually) the AAT entity for the color and then we can search for and use that value using our hybrid human/sparql interface. The nice thing about this is that then a machine processing the data downstream can reason about it make inferences...perhaps you could find all artworks across datasets that are red, or you could combine color values in a Sparql query and/or later when a machine is already recognizing some color aspects of a piece - it can likely be grouped using a linked data entity better than having text as as a value.

**Comment by aisaac** [comment]
And again Europeana is not going to help so much :-) Or maybe it will.
We do use both dimension and type for color: some objects from some collections come with a classification that is used in the 'channel' in our portal that is meant to provide a tailored access to these collections, see the 'colour' facet below:
https://www.europeana.eu/portal/en/collections/fashion?q=*%3A*&view=grid

For other 'channels' we have analyzed the images, stored that in a technical color (as dimension) field and use it for the 'Media/image' facet below
https://www.europeana.eu/portal/en/collections/art?q=%2A%3A%2A&f%5BTYPE%5D%5B%5D=IMAGE&view=grid

**Comment by zeroexp** [comment]
one note about color is that a machine can already provide very precise color information about regions - I think someone mentioned this in the call, where some DAM systems already have this is in use. Wouldn't one consideration for the future be to use a standard interoperability convention with a URI that could be compatible with whatever an AI machine might identify and be able to supply

**Comment by illip** [comment]
I have to say that I'm not a color expert, especially knowing that CHIN does not handle this type of information at the moment :p

From my point of view, it seems obvious, after reading your comments, that we need to manage colors as dimensions and types. In my opinion, the important thing is that the model allows us to cross-search the colors whatever modeling we used. So I think we need a standardized alignment between the very precise dimension point of view and the maybe more generic type one. I think we can't only give the choice to the cataloguer which type should be applied for a specific "calculated" color because it could become quite messy quickly. 

From my point of view, there is a part of the issue that have to be manage outside Linked.Art with a color vocabulary that reconciliates both approaches notably by handling the fuzziness between a label and a color spectrum. In other words, I think we need a way to reconcile both approaches and this goes through a well constructed vocabulary. 

So I don't know if there is such kind of product on the market but I think it would be interesting to be able to look for all "red" objects of a collection that means all the ones that are typed "red" (including all the "red" children) but also the ones that are in the "red spectrum" dimension. Conversely, someone who's looking for a specific color code could also find other objects that might fit his needs based on a more generic type.

**Comment by azaroth42** [comment]
```
{
  "type": "HumanMadeObject",
  "dimension": [
    {
      "type": "Dimension",
      "value": "FEFEFE",
      "unit": <aat:rgb-color-space>,
      // --- color profile here? 
      "classified_as": [ 
        {
          "id": "http://vocab.getty.edu/aat/whiteness",
          "type": "Type",
          "_label": "White"
        },
        {
          "id": "http://vocab.getty.edu/aat/colorness"
          "type": "Type",
          "_label": "Color"
        }
       ]
     }
  ]
}
```

**Comment by aisaac** [comment]
For the record, I've expressed doubts about this pattern as it puts the color dimension more prominent than the type. That may be acceptable. But we have cases where color will be available as type only and I don't think the pattern will look as good then. Also, the consequence of the pattern is also that in an RDF world the data could be interpreted in a way that a same value will carry forever and everywhere the type it was classified against once. I.e. FEFEFE would always be "white" and "color". This is dangerous when one considers that color classification is subjective and may depend from one object to the other.

Would it be possible to have a pattern where the two aspects of color are represented side by side (i.e. one is not predicated onto another)? Something like a blank node representing an 'colour attribute' that could be dimensioned and/or classified?

**Comment by azaroth42** [comment]
For the reuse question first, each FEFEFE dimension would have a different URI, even if it has the same value.  Dimensions (and Monetary Amounts) [in CIDOC-CRM are individual instances](http://lists.ics.forth.gr/pipermail/crm-sig/2017-January/002859.html), not a quantified type. So I can have a different 100 USD, or a different 36 inches. The use case is the dimension being the result of a measurement activity ... you might have measured something that is approximated to 36 inches, but that's an approximation of the real measurement the same way that FEFEFE is an approximation of the actual color. Meaning ... an institution can classify FEFEFE as Gray for one object and as White for another, with no inconsistency, as they're different instances.

I agree that it's a bit weird to have a blank Dimension sitting between the classification and the object ... but the alternatives all seem worse:

* We can't really go the other way from the object to type to dimension, as dimensions are meaningless without a reference resource.
* If we split them apart, then we won't know whether there are some categories and some separate measurements, or if they're overlapping. That would make the UI and search harder.
* If we don't allow dimensions, the conservation and automated analysis is impossible
* If we don't allow categories, then the curator description from a drop down is impossible
* We could introduce a new extension class, with all the costs of doing that, just to resolve the structure of the data having a less-than-ideal-but-not-wrong implication of precedence. That seems a very expensive solution.

**Comment by SamiNorling** [comment]
Our CMS data contains only a simple classification of color in our local thesaurus, e.g. "white", which I plan to reconcile with AAT in the future.

If I understand the previous conversation correctly, namely that the dimensions would be blank between the classification and the object, then my JSON-LD would look like:
```
{
  "type": "HumanMadeObject",
  "id": "https://data.discovernewfields.org/object/1234",
  "dimension": [
    {
      "type": "Dimension",
      "id": "https://data.discovernewfields.org/object/1234/color/1",
      "classified_as": [ 
        {
          "id": "https://data.discovernewfields.org/thesauri/color/white",
          "type": "Type",
          "_label": "white"
        },
        {
          "id": "http://vocab.getty.edu/aat/colorness"
          "type": "Type",
          "_label": "Color"
        }
       ]
     }
  ]
}
```
Is this a correct interpretation of the conversation? Whatever the recommended model is at this time, I will model with our data and provide as examples in our [Linked Art GitHub](https://github.com/IMAmuseum/LinkedArt).

**Comment by azaroth42** [comment]
Yes, that's correct :) It has a color dimension, which we know to be 'white' but not exactly its value.

**Comment by natuk** [comment]
Sorry, I am returning to this with a bit of a delay and apologies if I am misinterpreting the snippet above. My view is that the dimension does not have a type of colour. The feature of the object has a type of colour. Looking for red objects, should not require me to look for red dimensions. The type of dimension is always "colour", no?

**Comment by azaroth42** [comment]
Thanks Thanasis! I agree that the type of the dimension is always at least "color" (or "colour", yay for URIs!). 

To the issue of redness, the intent is to say that this particular dimension instance is not just any color, but a "red color".  Red Colors being a narrower concept than all colors, without being precise as to the exact value of red-ness. 

That said, if there's a better way to have a single pattern that encompasses both Dimension/Measurement and Categorization ... the ship hasn't sailed :)

**Comment by natuk** [comment]
Yes, easy for me to criticise! Should propose something better if I can...
I think the dimension type is "color" in the same way that elsewhere it is "height". I.e. the dimension measures only "color" (as in http://vocab.getty.edu/page/aat/300056130 ) and not "red color" (as in http://vocab.getty.edu/page/aat/300126225 ). I am not sure that it can be done in a single pattern. I describe something like this but I suspect it is not simple enough for the Linked.Art model:

```
{
  "type": "HumanMadeObject",
  "bears_feature" : [
    {
    "type": "Feature",
    "classified_as": [ 
        {
          "id": "http://vocab.getty.edu/aat/300126225",
          "type": "Type",
          "_label": "red"
        }
      ]
    }
  ],
  "dimension": [
    {
      "type": "Dimension",
      "value": "FF0000",
      "unit": "<aat:rgb-color-space>"
     }
  ]
}
```

**Comment by azaroth42** [comment]
To @natuk's version ... yeah, the challenge for this would be when there are are (say) five colors that are useful to record. The additional `Feature`s (which we don't use to date, cf the part discussion) would be rather overwhelming.

As an aside ... can features be entirely non-contiguous? All the bits of the painting that are red?  
Is there a "feature" of a book which is all of the printed parts of the pages?

To move forward with the issue, I'll document the "red dimension" version, but leave the issue open and we can discuss the feature (or just part) alternative.

**Comment by natuk** [comment]
> As an aside ... can features be entirely non-contiguous?

Yes, I think they can. See the "scattered fiat objects" like Hawai's map in Barry Smith's paper on "Fiat Objects". But I think that they all need to be on the same `E19 Physical Object`, i.e. I do not think that a `E26 Physical Feature` can be on two objects, which may be an issue for things like [this](https://www.tate.org.uk/art/artworks/finlay-the-world-has-been-empty-since-the-romans-t07531) where for example "M" is a feature split across two objects.

**Comment by cbutcosk** [comment]
Worth returning to this in light of #518 -- minimally the example in the model docs needs updated to use floats for `value` but also probably refactored to include an Identifier with a hex string for its `content` in `identified_by`, also possibly with a partition of the Dimension by Dimensions for the R, G, and B value for color components.

**Comment by azaroth42** [comment]
+1 to `identified_by` of the hex string, great call. Updated the docs, and closing done.

**Comment by azaroth42** [comment]
Closed by #587
